tasks/sgx.py

```diff
@@ -81,45 +94,50 @@ def install_sgxsdk():
         f.write("source {}/sgxddk/environment\n".format(SGX_INSTALL_DIR))
 
 
-@task
 def install_net_core_sdk():
-    run("apt update", shell=True)
-    run("apt install -y dotnet-sdk-3.1", shell=True)
+    run("apt update", shell=True, check=True)
+    run("apt install -y dotnet-sdk-3.1", shell=True, check=True)
 
 
 @task
-def install():
-    if dcap_driver_installed():
+def install(ctx):
+    """
+    Install SGX drivers
+    """
+    if is_driver_installed("dcap"):
         print("DCAP Driver installed. Skipping installation!")
     else:
         install_dcap()
 
-    if os.path.isfile("/opt/intel/sgxsdk/environment"):
+    if is_driver_installed("sgx"):
         print("SGX SDK installed. Skipping installation!")
     else:
         install_sgxsdk()
     install_net_core_sdk()
     clone_azure_attestation_repo()
 
 
-@task
-def generate_quotes():
+def demo_generate_quotes():
     run(
         "cd /opt/maa/intel.sdk.attest.sample/genquotes && AZDCAP_DEBUG_LOG_LEVEL=INFO bash ./runall.sh",
```

Comment:
Rather than doing `cd &&` in these commands you can specify a working directory with the `cwd` kwarg.